src/pygeoprocessing/routing/routing.pyx

```diff
@@ -796,35 +792,18 @@ def fill_pits(
                 f'{current_pixel} of {raster_x_size * raster_y_size} '
                 'pixels complete')
 
-        # make a buffer big enough to capture block and boundaries around it
-        dem_buffer_array = numpy.empty(
-            (offset_dict['win_ysize']+2, offset_dict['win_xsize']+2),
-            dtype=numpy.float64)
-        dem_buffer_array[:] = dem_nodata
-
-        # attempt to expand read block by a pixel boundary
-        (xa, xb, ya, yb), modified_offset_dict = _generate_read_bounds(
-            offset_dict, raster_x_size, raster_y_size)
-        filled_dem_managed_raster.flush()
-        target_dem_raster = gdal.OpenEx(target_filled_dem_raster_path)
-        target_dem_band = target_dem_raster.GetRasterBand(1)
-        dem_buffer_array[ya:yb, xa:xb] = target_dem_band.ReadAsArray(
-                **modified_offset_dict).astype(numpy.float64)
-        target_dem_band = None
-        target_dem_raster = None
-
         # search block for locally undrained pixels
         for yi in range(1, win_ysize+1):
+            yi_root = yi-1+yoff
             for xi in range(1, win_xsize+1):
-                center_val = dem_buffer_array[yi, xi]
+                xi_root = xi-1+xoff
+                center_val = filled_dem_managed_raster.get(xi_root, yi_root)
                 if _is_close(center_val, dem_nodata, 1e-8, 1e-5):
                     continue
 
                 # this value is set in case it turns out to be the root of a
                 # pit, we'll start the fill from this pixel in the last phase
                 # of the algorithm

```

Comment:
Does this comment still make sense here?

Reply:
Ahh no it doesn't. I scooted it up a couple of lines in b00244f